### DISCUSSION TITLE
psycopg2 - no available wheel for linux platforms ?

### DISCUSSION THREAD

**Discussion by lb-merc** [psycopg/psycopg2#1661] [Q&A]
Hello there :)

We are trying to install psycopg2 for one of our app (in python 3.11) and we are looking for (but fail to find) a wheel created for linux platforms . In pypi, we only find wheels designed for windows platforms.

- As it is needed to run on production env, we are advised to use psycopg2 , not psycopg2-binary (as advised in the docs). Too bad as , as there was a linux wheel for the -binary package ( :) ).

- We can not use yet the new version of psycopg ( psycopg3)  as it seems it is relatively quite new and still do not have all "older" features ( for instance, "extensions" part that was in psycopg2) .

- And we have to build our packaging from a wheel as we have to repackage it to onboard also external libs (for instance, libpq not available by default in the OS running the process and not possible to "manually" install it on the OS; aws lambda topic , I won't detail that much).

Would it be possible to have wheels created for linux platforms ( more specifically , what we are searching is for "manylinux2014_x86_64" platform)    ? Do you have some hints to help us ?

Thanks in advance for your help

**Comment by dvarrazzo** [comment]
> We are trying to install psycopg2 for one of our app (in python 3.11) and we are looking for (but fail to find) a wheel created for linux platforms . In pypi, we only find wheels designed for windows platforms.

Yes, because wheels were problematic initially and pip doesn't offer an opt-in way to install binary packages, only an opt-out one.

> - As it is needed to run on production env, we are advised to use psycopg2 , not psycopg2-binary (as advised in the docs). Too bad as , as there was a linux wheel for the -binary package ( :) ).

Yes, that's our own-made opt-in mechanism.

> - We can not use yet the new version of psycopg ( psycopg3)  as it seems it is relatively quite new and still do not have all "older" features ( for instance, "extensions" part that was in psycopg2) .

Quite new? Psycopg 3 has been released more than 2 years ago and has more than 30 releases. The only features it's missing are Large Objects support and replication, but it has many more features and improvements, including more straightforward packaging. See the [comparison grid](https://www.psycopg.org/features/).

> - And we have to build our packaging from a wheel as we have to repackage it to onboard also external libs (for instance, libpq not available by default in the OS running the process and not possible to "manually" install it on the OS; aws lambda topic , I won't detail that much).

> Would it be possible to have wheels created for linux platforms ( more specifically , what we are searching is for "manylinux2014_x86_64" platform)    ? Do you have some hints to help us ?

Not from the psycopg2 project. You can make them yourself by following what the CI does: it's everything in the project support scripts.

If you really want them I can set up your CI to create them as a form of paid consultancy. Feel free to reach me in private if you want so.
